### ISSUE TITLE
Using  TargetType="{x:Type Button} causes compilation error in Setters of VisualState 

### ISSUE THREAD

**Issue by sbll885** [dotnet/maui#21270]
### Description

Take a standard Button Style and use "{x:Type Button}" instead of "Button" for TargetType.

```xml
<Style TargetType="{x:Type Button}">
		<Setter Property="TextColor" Value="{AppThemeBinding Light={StaticResource White}, Dark={StaticResource PrimaryDarkText}}" />
		<Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource PrimaryDark}}" />
		<Setter Property="FontFamily" Value="OpenSansRegular"/>
		<Setter Property="FontSize" Value="14"/>
		<Setter Property="BorderWidth" Value="0"/>
		<Setter Property="CornerRadius" Value="8"/>
		<Setter Property="Padding" Value="14,10"/>
		<Setter Property="MinimumHeightRequest" Value="44"/>
		<Setter Property="MinimumWidthRequest" Value="44"/>
		<Setter Property="VisualStateManager.VisualStateGroups">
			<VisualStateGroupList>
				<VisualStateGroup x:Name="CommonStates">
					<VisualState x:Name="Normal" />
					<VisualState x:Name="Disabled">
						<VisualState.Setters>
							<Setter Property="TextColor" Value="{AppThemeBinding Light={StaticResource Gray950}, Dark={StaticResource Gray200}}" />
							<Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource Gray600}}" />
						</VisualState.Setters>
					</VisualState>
					<VisualState x:Name="PointerOver" />
				</VisualStateGroup>
			</VisualStateGroupList>
		</Setter>
	</Style>
```

### Steps to Reproduce

1.  Create new MAUI application
2.  Update Button style in  Styles.xaml by changing Style TargetType to "{x:Type Button}" instead of "Button"
3.  Try compile. 
4.  Observe XamlC error XFC0040: Cannot convert value "TextColor" to "Microsoft.Maui.Controls.BindableProperty".dot

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

I was *not* able test on other platforms

### Did you find any workaround?

Use "Button" instead. Not sure how should it work.

**Comment by jfversluis** [comment]
Coincidence or not, this seems to be fixed by #20915 which was just merged. Coming to a next release!

**Comment by blackforest-tom** [comment]
I just updated to 8.0.20 and I get this exact XC0040 for the first time. I have styled everything in the "{x:Type ...}" way, and now I cannot compile my app anymore. Before, using 8.0.14, that worked without any issues. That's pretty strange, because its in the release notes that both should be allowed starting from 8.0.20... Any thoughts on this? @jfversluis 

![grafik](https://github.com/dotnet/maui/assets/45143782/d105d170-0a91-4b34-93b6-c208618f053e)
